xds/internal/xdsclient/xds.go

```diff
@@ -323,8 +323,35 @@ func processNetworkFilters(filters []*v3listenerpb.Filter) ([]HTTPFilter, error)
 			}
 			if !seenHCM {
 				// TODO: Implement terminal filter logic, as per A36.
-				httpFilters = filters
+				filterChain.HTTPFilters = filters
 				seenHCM = true
+				switch hcm.RouteSpecifier.(type) {
+				case *v3httppb.HttpConnectionManager_Rds:
+					if hcm.GetRds().GetConfigSource().GetAds() == nil {
+						return nil, fmt.Errorf("ConfigSource is not ADS: %+v", hcm)
+					}
+					name := hcm.GetRds().GetRouteConfigName()
+					if name == "" {
+						return nil, fmt.Errorf("empty route_config_name: %+v", hcm)
+					}
+					filterChain.RouteConfigName = name
+				case *v3httppb.HttpConnectionManager_RouteConfig:
+					// "RouteConfiguration validation logic inherits all
+					// previous validations made for client-side usage as RDS
+					// does not distinguish between client-side and
+					// server-side." - A36
+					// Can specify v3 here, as will never get to this function
+					// if v2.
+					routeU, err := generateRDSUpdateFromRouteConfiguration(hcm.GetRouteConfig(), nil, false)
+					if err != nil {
+						return nil, fmt.Errorf("failed to parse inline RDS resp: %v", err)
+					}
+					filterChain.InlineRouteConfig = &routeU // Now that all this is here, scale up listener inputs in test to persist route update
```

Comment:
Nit: This comment is long enough that it can live on a line on its own. Eitherways, it is not super clear what this comment is trying to convey.

Reply:
Deleted.